upcloud/problem.go

```diff
@@ -34,8 +42,18 @@ func (p *Problem) Error() string {
 	return sb.String()
 }
 
-// ProblemInvalidParam is a type describing extra information in the Problem type's InvalidParams field.
-type ProblemInvalidParam struct {
-	Name   string `json:"name"`
-	Reason string `json:"reason"`
+// ErrorCode returns a short string that identifies the error; it should be used for programmatic comparisons
+func (p *Problem) ErrorCode() string {
+	// Type is a URL, we need to extract meaningful fragment from it for comparison purposes
+	if strings.HasPrefix(p.Type, "https://") {
+		parsedURL, err := url.Parse(p.Type)
+		if err != nil {
+			return ""
+		}
+
+		return strings.Replace(parsedURL.Fragment, "ERROR_", "", 1)
+	}
+
+	// Type is just bare error code string (this happens when error returned from the API is not in the json+problem format)
+	return p.Type
```

Comment:
Would the below suggestion work instead of the HTTPS scheme check?

It's a given that we have to make many assumptions. It could be nice to remove one of those: depending on exact URL scheme.

```suggestion
	// If the type is a URL, we need to extract meaningful fragment from it for comparison purposes
	parsedURL, err := url.Parse(p.Type)
	if err != nil {
		// Type is just bare error code string (this happens when error returned from the API is not in the json+problem format)
		return p.Type
	}

	return strings.Replace(parsedURL.Fragment, "ERROR_", "", 1)
```

Reply:
Funny enough, `url.Parse("GROUP_NOT_FOUND")` does not return an error :P Need to also check for empty host or scheme. But overall good suggestion, added!